shell/platform/darwin/macos/framework/Source/FlutterVersions.mm

```diff
@@ -0,0 +1,65 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterVersions.h"
+
+bool isOperatingSystemAtLeastVersion(NSInteger majorVersion,
+                                     NSInteger minorVersion,
+                                     NSInteger patchVersion) {
+  NSCAssert(majorVersion != -1, @"majorVersion must not be ignored");
+  NSCAssert(minorVersion != -1 || patchVersion == -1,
+            @"if minorVersion is ignored, patchVersion must also be ignored");
+  NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
```

Comment:
As noted in the last review, `NSProcessInfo` has `isOperatingSystemAtLeast`.

In conjunction with the other comments (zero being fine as an unset value for >=, and <= being easily expressed with a different >=) that should mean that this entire file can be completely eliminated in favor of just using an existing OS method.

(And then I don't have to ask you where the tests for these functions are 😉 )

Reply:
ah you are right